lib/src/web_socket.dart

```diff
@@ -144,6 +144,14 @@ class WebSocket {
   /// The WebSocket [Connection].
   Connection get connection => _connectionController;
 
+  /// The subprotocol used by the connection
+  ///
+  /// The subprotocol is sent by the server during the initial
+  /// connection handshake, if the client and server aren't
+  /// able to negotiate a protocol during the handshake, it will
+  /// return `null`
+  String? get protocol => _channel?.protocol;
```

Comment:
Hi, I was just looking at implementing tests, I'm wondering which group you think it belongs in, since it's related to both connection and messages

Reply:
I'm working on adding this test, but I'm running into a strange issue with the `WebSocketTransformer` in the `createWebSocketServer()` method, for some reason, in the `protocolSelector` parameter for the transformer, the passed in list of protocols all have the first letter missing for some reason, not sure if you know where that issue could be from, since it's causing the tests to fail in a very strange way

Reply:
hmm not sure but happy to take a look later today